flink-connectors/flink-connector-aws-base/src/test/java/org/apache/flink/connector/aws/testutils/LocalstackContainer.java

```diff
@@ -62,7 +62,7 @@ public String getEndpoint() {
         @Override
         protected void waitUntilReady() {
             try {
-                Thread.sleep(10000);
+                Thread.sleep(30_000);
```

Comment:
Please justify the reason for this.

Is there any other way we can establish if the test stack is ready?

Reply:
I couldn't see a good way to poll whether the container is ready. On the `testcontainers` side, I tried tracking the output of `isRunning()`/`isCreated()`/`isReusable()` through successful and unsuccessful runs and none of them are telling. Anything on the `localstack` api side puts it in a funny state in the same way the s3bucket read strategy does.

Reply:
There is a solution here: https://ignas.me/tech/waiting-localstack-s3-start/

> I’ve noticed that hitting Refresh button in LocalStack Dashboard calls /graph endpoint to know if and how many nodes (mock services) are available, i.e. healthy and ready to accept incoming requests. Bullseye!

Reply:
Thanks, I have created this: https://issues.apache.org/jira/browse/FLINK-26309.